ui/app/src/views/ViewDashboard.tsx

```diff
@@ -21,16 +25,54 @@ const DEFAULT_DASHBOARD_ID = 'node-exporter-full';
  * The View for viewing a Dashboard.
  */
 function ViewDashboard() {
-  const dashboard = useSampleData<DashboardResource>(
-    new URLSearchParams(window.location.search).get('dashboard') || DEFAULT_DASHBOARD_ID
-  );
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const dashboardParam = searchParams.get('dashboard');
+  const dashboard = useSampleData<DashboardResource>(dashboardParam || DEFAULT_DASHBOARD_ID);
+  const defaultDuration = dashboard?.spec.duration ?? '1h';
+
+  const fromParam = searchParams.get('from');
+  const toParam = searchParams.get('to');
+  const parsedParam = fromParam !== null ? fromParam.split('-')[1] : defaultDuration;
+  const pastDuration = parsedParam && isDurationString(parsedParam) ? parsedParam : defaultDuration;
+
+  // TODO: cleanup, better way to determine query param format
+  const defaultTimeRange =
+    fromParam !== null && toParam !== null && toParam !== 'now'
+      ? { start: new Date(Number(fromParam)), end: new Date(Number(toParam)) }
+      : { pastDuration: pastDuration };
+  const [activeTimeRange, setActiveTimeRange] = useState<TimeRangeValue>(defaultTimeRange as TimeRangeValue);
 
   // TODO: Loading indicator
   if (dashboard === undefined) {
     return null;
   }
 
-  return <DashboardView dashboardResource={dashboard} />;
+  const handleOnDateRangeChange = (event: TimeRangeValue) => {
+    // TODO: refactor, preserve all existing params
+    if (isRelativeValue(event)) {
+      setSearchParams({
```

Comment:
This PR isn't ready to review yet, I'm planning to move this to the other ViewDashboard.tsx inside the dashboards package and use this [query-string package](https://www.npmjs.com/package/query-string) instead of having separate implementations for React Router v5 and v6. This will allow for a more generalizable solution that we can also make use of for template variables. 



useQueryStringState

Reply:
See PR description--discussed this with Luke and decided to stick with the onDateRangeChange approach in this PR for now and add a better long term approach in the future using either a new QueryStringProvider or adding a queryParamOptions prop (this would also need to work with apps that use react router v5 and not break the back button / useLocation history)